src/xdsl/dialects/memref.py

```diff
@@ -207,7 +208,7 @@ class Global(Operation):
     # constant = AttributeDef(UnitAttr)
 
     @staticmethod
-    def get(sym_name,
+    def get(sym_name: StringAttr,
```

Comment:
sym_name can also be a str

Reply:
Does build accept this? Or should I create a StringAttr from a potentially passed string?

Reply:
build accepts additionally to the attribute every parameters that can be passed to a builder.
So since we have `StringAttr.get(s: str)` marked as builder, it will accept both `str` and `StringAttr`.
So here you don't have to change anything besides the type signature ;)